bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/cache/ExpiringCacheMap.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.cache;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a simple expiring and reloading multiple key-value-pair cache implementation. The value expires after the
+ * specified duration has passed since the item was created, or the most recent replacement of the value.
+ *
+ * @author Christoph Weitkamp - Initial contribution and API.
+ *
+ * @param <K> the type of the key
+ * @param <V> the type of the value
+ */
+public class ExpiringCacheMap<K, V> {
+
+    private final Logger logger = LoggerFactory.getLogger(ExpiringCacheMap.class);
+
+    private final long expiry;
+    protected ConcurrentMap<K, ExpiringCache<V>> items;
+
+    /**
+     * Creates a new instance.
+     * 
+     * @param expiry the duration in milliseconds for how long the value stays valid
+     */
+    public ExpiringCacheMap(long expiry) {
+        this.expiry = expiry;
+        this.items = new ConcurrentHashMap<K, ExpiringCache<V>>();
+    }
+
+    /**
+     * Creates an {@link ExpiringCache} and adds it to the cache.
+     * 
+     * @param key the key with which the specified value is to be associated
+     * @param action the action for the item to be associated with the specified key to retrieve/calculate the value
+     * @return the value associated with the given key
+     */
+    public V put(K key, Supplier<V> action) {
+        return put(key, new ExpiringCache<V>(expiry, action));
+    }
+
+    /**
+     * Adds an {@link ExpiringCache} to the cache.
+     * 
+     * @param key the key with which the specified value is to be associated
+     * @param item the item to be associated with the specified key
+     * @return the value associated with the given key
+     */
+    public V put(K key, ExpiringCache<V> item) {
+        if (key == null) {
+            throw new IllegalArgumentException("Item cannot be added as key is null.");
+        }
+        if (item == null) {
+            throw new IllegalArgumentException("Item cannot be added as item is null.");
+        }
+
+        items.put(key, item);
+
+        return item.getValue();
```

Comment:
I don't know if the `put` method should trigger a cache refresh (that could be slow).